crates/net/network/src/network.rs

```diff
@@ -152,6 +153,12 @@ impl NetworkHandle {
         rx.await
     }
 
+    /// Toggles network connection state between [`NetworkConnectionState::Hibernate`] and
+    /// [`NetworkConnectionState::Active`].
+    pub async fn toggle_network_conn(&self, network_conn: NetworkConnectionState) {
+        self.send_message(ToggleNetworkConnection(network_conn));
+    }
```

Comment:
here we want to flatten the enum into individual commands, be we can keep this as well, the other calls simply call this then